addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -201,8 +204,12 @@ private void updateThingFromDevice(Thing thing, DeviceModel device) {
                         calendar.setTime(new Date(device.getHkr().getNextchange().getEndperiod() * 1000L));
                         updateThingChannelState(thing, CHANNEL_NEXTCHANGE, new DateTimeType(calendar));
                     }
-                    updateThingChannelState(thing, CHANNEL_NEXTTEMP,
-                            new DecimalType(HeatingModel.toCelsius(device.getHkr().getNextchange().getTchange())));
+                    if (HeatingModel.TEMP_FRITZ_UNDEFINED.equals(device.getHkr().getNextchange().getTchange())) {
```

Comment:
Should you maybe put this logic in a method it seems like something you could need more often.